shared/django_apps/user_measurements/partitioning.py

```diff
@@ -0,0 +1,27 @@
+from dateutil.relativedelta import relativedelta
+from psqlextra.partitioning import (
+    PostgresCurrentTimePartitioningStrategy,
+    PostgresPartitioningManager,
+    PostgresTimePartitionSize,
+)
+from psqlextra.partitioning.config import PostgresPartitioningConfig
+
+from shared.django_apps.user_measurements.models import UserMeasurement
+
+# Overlapping partitions will cause errors - https://www.postgresql.org/docs/current/ddl-partitioning.html#DDL-PARTITIONING-DECLARATIVE -> "create partitions"
+manager = PostgresPartitioningManager(
+    [
+        # 12 partitions ahead, each partition is 1 month
+        # Partitions can be deleted after 12 months of their starting date, not their creation, via the pgpartition command.
+        # They won't be automatically deleted though.
+        # Partitions will be named `[table_name]_[year]_[3-letter month name]`.
+        PostgresPartitioningConfig(
+            model=UserMeasurement,
+            strategy=PostgresCurrentTimePartitioningStrategy(
+                size=PostgresTimePartitionSize(months=1),
+                count=12,
+                max_age=relativedelta(months=12),
```

Comment:
is this not insinuated by the count above?

Reply:
Yeah this settings is mostly a "after how long are these partitions safely deletable by the partition command". So these could be 24 or any non 12 number for instance, so while they're the same number, they are distinct. This also means that the April 2024 partition is deletable in april 2025, and the may 2024 one in may 2025 and so on